### PR TITLE
[MM-29245] Add Billing SubSection

### PR DESCRIPTION
**Pull request by hahmadia**
#### Summary

Add Billing SubSections

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-29245

#### Related PR's
**Billing:**

- [ ] Webapp: https://github.com/mattermost/mattermost-webapp/pull/7673
- [ ]  Server: https://github.com/mattermost/mattermost-server/pull/17107
- [ ] Redux: https://github.com/mattermost/mattermost-redux/pull/1405

### **HUGE THANK YOU** to Maria from the Cloud team for helping me out with figuring out these permissions.

### CODE REVIEW

File: components/admin_console/admin_definition.jsx [hunk 1]
```diff
@@ -285,7 +290,7 @@ const AdminDefinition = {
                 id: 'BillingHistory',
                 component: BillingHistory,
             },
-            isDisabled: it.not(it.userHasWritePermissionOnResource('billing')),
+            isDisabled: it.not(it.userHasWritePermissionOnResource(RESOURCE_KEYS.BILLING.BILLING_HISTORY)),
```
[the remark this answers is not in the document]
Comment (hahmadia) [reply components/admin_console/admin_definition.jsx:295]: Apologies. That's a mistake on my end. It should be disabled if you don't have write permissions and hidden if you don't have read permission.

I had forgot to add the `isHidden` to the subsections ... oopsies.

Thanks for the catch!

File: components/admin_console/admin_definition.jsx [hunk 2]
```diff
@@ -224,7 +226,9 @@ const AdminDefinition = {
         sectionTitleDefault: 'About',
         isHidden: it.any(
             it.configIsTrue('ExperimentalSettings', 'RestrictSystemAdmin'),
-            it.not(it.userHasReadPermissionOnResource('about')),
+            it.all(
+                it.not(it.userHasReadPermissionOnResource(RESOURCE_KEYS.ABOUT.EDITION_AND_LICENSE)),
```
Comment (marianunez) [inline components/admin_console/admin_definition.jsx:230]: Are these `About` changes related to this PR?

**Review by marianunez — APPROVED** [review]
Account and Billing sections LGTM. However, I see other changes in this PR that are unrelated which you probably need another reviewer for

File: components/admin_console/admin_definition.jsx (hunk 2 above)
Comment (hahmadia) [reply components/admin_console/admin_definition.jsx]: Sorry that should be gone now

**Comment by hahmadia** [comment]
**UPDATE**

We had a meeting regarding this particular section and it was decided that we will leave it as it is for the following two reasons:

1. The interdependency of sections makes things a bit more complicated and not something we need to address at this very moment since customers haven't brought it up or asked for it.
2. Maria mentioned that they are still in the early stages of the billing section and have plans to change things which means that our effort for this could potentially go away.

Until required, we will leave Billing as it is and come back in the future once it's in a more stable position.

**Pull request closed without merging**